src/main/java/org/fenixedu/academic/domain/accounting/events/gratuity/GratuityEventWithPaymentPlan.java

```diff
@@ -268,7 +268,7 @@ public GratuityWithPaymentPlanPR getPostingRule() {
     }
 
     @Override
-    public Map<LocalDate, Money> calculateDueDateAmountMap(DateTime when) {
+    public Map<LocalDate, Money> calculateDueDateAmountMap() {
         return getGratuityPaymentPlan().getInstallmentsSet().stream().filter( i -> i.calculateBaseAmount(this).greaterThan(Money.ZERO))
                    .collect(Collectors.toMap(i -> i.getEndDate(this), i -> i.calculateBaseAmount(this)));
```

Comment:
The map entries should be combined using the Money::add method to avoid DuplicateKey errors.
If the due date coincides, then for the due date amount map it is effectively the same entry.

`    @Override
    public Map<LocalDate, Money> calculateDueDateAmountMap() {
        return getGratuityPaymentPlan().getInstallmentsSet().stream()
                .filter(i -> i.calculateBaseAmount(this).greaterThan(Money.ZERO))
                .collect(Collectors.toMap(i -> i.getEndDate(this), i -> i.calculateBaseAmount(this), Money::add));
    }`